lib/active_merchant/billing/gateways/mundipagg.rb

```diff
@@ -0,0 +1,287 @@
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class MundipaggGateway < Gateway
+      self.live_url = 'https://api.mundipagg.com/core/v1/'
+
+      self.supported_countries = ['US']
+      self.default_currency = 'USD'
+      self.supported_cardtypes = [:visa, :master, :american_express, :discover]
+
+      self.homepage_url = 'https://www.mundipagg.com/'
+      self.display_name = 'Mundipagg'
+
+      STANDARD_ERROR_CODE_MAPPING = {
+        '400' => STANDARD_ERROR_CODE[:processing_error],
+        '401' => STANDARD_ERROR_CODE[:config_error],
+        '404' => STANDARD_ERROR_CODE[:processing_error],
+        '412' => STANDARD_ERROR_CODE[:processing_error],
+        '422' => STANDARD_ERROR_CODE[:processing_error],
+        '500' => STANDARD_ERROR_CODE[:processing_error]
+      }
+
+      STANDARD_ERROR_MESSAGE_MAPPING = {
+        '400' => 'Invalid request;',
+        '401' => 'Invalid API key;',
+        '404' => 'The requested resource does not exist;',
+        '412' => 'Valid parameters but request failed;',
+        '422' => 'Invalid parameters;',
+        '500' => 'An internal error occurred;'
+      }
+
+      def initialize(options={})
+        requires!(options, :api_key)
```

Comment:
I noticed that the `fixtures.yml` contains an entry for `gateway_id`. Any particular reason we don't require `gateway_id` here?

Reply:
Its a field they use the track with. They only asked us to include it when were trying to debug how to send the voucher requests correctly.

Reply:
I don't think its just for testing. I think it similar to when we want to monitor traffic from a specific merchant. Or when a gateway wants to track Spreedly traffic.